pstl/include/pstl/internal/algorithm_impl.h

```diff
@@ -1752,25 +1758,28 @@ __brick_is_partitioned(_RandomAccessIterator __first, _RandomAccessIterator __la
     }
 }
 
-template <class _ExecutionPolicy, class _ForwardIterator, class _UnaryPredicate, class _IsVector>
+template <class _Tag, class _ExecutionPolicy, class _ForwardIterator, class _UnaryPredicate>
 bool
-__pattern_is_partitioned(_ExecutionPolicy&&, _ForwardIterator __first, _ForwardIterator __last, _UnaryPredicate __pred,
-                         _IsVector __is_vector, /*is_parallel=*/std::false_type) noexcept
+__pattern_is_partitioned(_Tag, _ExecutionPolicy&&, _ForwardIterator __first, _ForwardIterator __last,
+                         _UnaryPredicate __pred) noexcept
 {
-    return __internal::__brick_is_partitioned(__first, __last, __pred, __is_vector);
+    return __internal::__brick_is_partitioned(__first, __last, __pred, typename _Tag::_IsVector{});
 }
 
-template <class _ExecutionPolicy, class _RandomAccessIterator, class _UnaryPredicate, class _IsVector>
+template <class _IsVector, class _ExecutionPolicy, class _RandomAccessIterator, class _UnaryPredicate>
 bool
-__pattern_is_partitioned(_ExecutionPolicy&& __exec, _RandomAccessIterator __first, _RandomAccessIterator __last,
-                         _UnaryPredicate __pred, _IsVector __is_vector, /*is_parallel=*/std::true_type)
+__pattern_is_partitioned(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec, _RandomAccessIterator __first,
+                         _RandomAccessIterator __last, _UnaryPredicate __pred)
 {
     if (__first == __last)
     {
         return true;
     }
     else
     {
+        using __backend_tag = typename decltype(__tag)::__backend_tag;
+        auto __is_vector = _IsVector{};
```

Comment:
You may construct it in-place everywhere you need that. Thus, you don't need a capture.

I would prefer to have to consistent approach. This comment is applicable to other places.

Reply:
Not quite agree. 
Of course, everyone it can be in-placed. But, I did it deliberately to minimize the changes, especially for more comfortable review process by guys from LLVM.

Reply:
The goal is not to minimize the changes. They are already huge. I believe we need to avoid capturing when we can. But this is really small thing.

I am more concerned about consistency in terms of one patch to be submitted in LLVM.